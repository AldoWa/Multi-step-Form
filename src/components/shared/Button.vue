<template>
  <button 
    :class="cn(buttonVariants({ variant: props.variant, class: props.class }))"
    :type="type"
  >
    <slot></slot>
  </button>
</template>

<script setup lang="ts">
import { cva } from "class-variance-authority"
import { ButtonHTMLAttributes } from "vue";
import { cn } from "../../utils";


const buttonVariants = cva(
  "py-[10px] px-[26px] rounded-lg font-medium text-sm focus:outline-none focus:ring-2 focus:ring-[#FC6C4C] focus:ring-opacity-50 transition-all duration-200 ease-in-out",
  {
    variants: {
      variant: {
        default: "bg-[#FC6C4C] text-white",
        outline: "bg-transparent border-2 border-[#FC6C4C] text-[#FC6C4C]",
      },
    },
    defaultVariants: {
      variant: "default",
    }
  }
) 

type ButtonProps = {
  variant?: 'default' | 'outline';
  class?: ButtonHTMLAttributes['class'];
  type?: ButtonHTMLAttributes['type'];
}; 

const props = withDefaults(defineProps<ButtonProps>(), {
  variant: 'default',
  type: 'button',
});
</script>
