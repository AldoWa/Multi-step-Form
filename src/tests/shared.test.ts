import { mount } from '@vue/test-utils'

import { it, describe, expect } from 'vitest'

import Button from '../components/shared/Button.vue'

describe('Shared', () => {
  describe('Button', () => {
    it('Should render the button with default variant and with "Button" span', () => {
      const button = mount(Button, {
        slots: {
          default: 'Button'
        }
      })

      const buttonElement = button.get('[data-testid="button"]')

      expect(buttonElement.classes()).toContain('bg-[#FC6C4C]')
      expect(buttonElement.text()).toBe('Button')
    })

    it('Should render the button with outline variant and with "Outline" span', () => {
      const button = mount(Button, {
        slots: {
          default: 'Outline'
        },
        props: {
          variant: 'outline', 
        }
      })

      const buttonElement = button.get('[data-testid="button"]')

      expect(buttonElement.classes()).toContain('bg-transparent')
      expect(buttonElement.text()).toBe('Outline')
    })

    it('Should match with the snapshot', () => {
      const button = mount(Button, {
        slots: {
          default: 'Button'
        }
      })

      expect(button.html()).toMatchSnapshot()
    })
  }) 
})