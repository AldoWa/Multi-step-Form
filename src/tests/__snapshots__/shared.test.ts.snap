// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shared > Button > Should match with the snapshot 1`] = `"<button class="py-[10px] px-[26px] rounded-lg font-medium text-sm focus:outline-none focus:ring-2 focus:ring-[#FC6C4C] focus:ring-opacity-50 transition-all duration-200 ease-in-out bg-[#FC6C4C] text-white" type="button" data-testid="button">Button</button>"`;
